test/e2e/util_gatewayapi_test.go

```diff
@@ -701,79 +701,69 @@ type testListener struct {
 func assertExpectedDNSRecords(t *testing.T, expectations map[expectedDnsRecord]bool) error {
 	t.Helper()
 
-	var expectationsMet bool
-
-	err := wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 1*time.Minute, false, func(context context.Context) (bool, error) {
-		haveExpectNotPresent := false
-		// expectationsMet starts true and gets set to false when some expectation is not met.
-		expectationsMet = true
-
-		dnsRecords := &v1.DNSRecordList{}
-		if err := kclient.List(context, dnsRecords, client.InNamespace(operatorcontroller.DefaultOperandNamespace)); err != nil {
-			return false, fmt.Errorf("failed to list DNSRecords: %v", err)
-		}
-
-		// Iterate over all expectations.
-		for exp, shouldBePresent := range expectations {
-			if !shouldBePresent {
-				haveExpectNotPresent = true
+	return wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 1*time.Minute, false,
+		func(ctx context.Context) (bool, error) {
+			dnsRecords := &v1.DNSRecordList{}
+			if err := kclient.List(ctx, dnsRecords, client.InNamespace(operatorcontroller.DefaultOperandNamespace)); err != nil {
+				return false, fmt.Errorf("failed to list DNSRecords: %v", err)
```

Comment:
In most of the polling cases we log the error and retry the API operator. Returning an error would break the polling loop.
```suggestion
			if err := kclient.List(ctx, dnsRecords, client.InNamespace(operatorcontroller.DefaultOperandNamespace)); err != nil {
			        t.Logf("Failed to list DNSRecords: %v", err)
				return false, nil
```